include/itkTorchImageToImageFilter.h

```diff
@@ -19,60 +19,88 @@
+
+  ITK_DISALLOW_COPY_AND_ASSIGN( TorchImageToImageFilter );
 
-  static constexpr unsigned int InputImageDimension = TInputImage::ImageDimension;
+  static constexpr unsigned int InputTorchImageDimension = TInputTorchImage::ImageDimension;
   static constexpr unsigned int OutputImageDimension = TOutputImage::ImageDimension;
 
-  using InputImageType = TInputImage;
+  /** Some convenient type aliases. */
+  using InputTorchImageType = TInputTorchImage;
+  using InputTorchImagePointer = typename InputTorchImageType::Pointer;
+  using InputTorchImageConstPointer = typename InputTorchImageType::ConstPointer;
+  using InputTorchImageRegionType = typename InputTorchImageType::RegionType;
+  using InputTorchImageRegionIterator = ImageRegionIterator< InputTorchImageType >;
+  using InputTorchImageRegionConstIterator = ImageRegionConstIterator< InputTorchImageType >;
+  using InputTorchImagePixelType = typename InputTorchImageType::PixelType;
+  using InputTorchImageSizeType = Size< InputTorchImageDimension >;
+
   using OutputImageType = TOutputImage;
-  using InputPixelType = typename InputImageType::PixelType;
-  using OutputPixelType = typename OutputImageType::PixelType;
+  using OutputImagePointer = typename OutputImageType::Pointer;
+  using OutputImageConstPointer = typename OutputImageType::ConstPointer;
+  using OutputImageRegionType = typename OutputImageType::RegionType;
+  using OutputImageRegionIterator = ImageRegionIterator< OutputImageType >;
+  using OutputImageRegionConstIterator = ImageRegionConstIterator< OutputImageType >;
+  using OutputImagePixelType = typename OutputImageType::PixelType;
+  using OutputImageSizeType = Size< OutputImageDimension >;
 
   /** Standard class typedefs. */
-  using Self = TorchImageToImageFilter<InputImageType, OutputImageType>;
-  using Superclass = ImageToImageFilter<InputImageType, OutputImageType>;
-  using Pointer = SmartPointer<Self>;
-  using ConstPointer = SmartPointer<const Self>;
+  using Self = TorchImageToImageFilter< InputTorchImageType, OutputImageType >;
+  using Superclass = ImageToImageFilter< InputTorchImageType, OutputImageType >;
+  using Pointer = SmartPointer< Self >;
+  using ConstPointer = SmartPointer< const Self >;
 
   /** Run-time type information. */
-  itkTypeMacro(TorchImageToImageFilter, ImageToImageFilter);
+  itkTypeMacro( TorchImageToImageFilter, ImageToImageFilter );
 
   /** Standard New macro. */
-  itkNewMacro(Self);
+  itkNewMacro( Self );
 
 protected:
-  TorchImageToImageFilter();
+  TorchImageToImageFilter() = default;
   ~TorchImageToImageFilter() override = default;
 
-  void PrintSelf(std::ostream & os, Indent indent) const override;
+  void PrintSelf( std::ostream &os, Indent indent ) const override;
+
+  void GenerateInputRequestedRegion() override;
 
-  using OutputRegionType = typename OutputImageType::RegionType;
+  void BeforeThreadedGenerateData() override;
```

Comment:
Overriding `GenerateData` is simpler and sufficient.

Reply:
Will do.